readme.md

```diff
@@ -541,80 +541,696 @@ namespace Azure.Service.Models
+
+``` C#
+// Generated/Models/Cat.Serialization.cs
+namespace Azure.Service.Models
+{
+    public partial class Cat
+    {
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName("name"u8);
+            writer.WriteStringValue(Name);
+            if (Optional.IsDefined(Color))
+            {
+                writer.WritePropertyName("color"u8);
+                writer.WriteStringValue(Color);
+            }
+            writer.WriteEndObject();
+        }
+
+        internal static Cat DeserializeCat(JsonElement element)
+        {
+            string color = default;
+            string name = default;
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("color"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    color = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("name"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    name = property.Value.GetString();
+                    continue;
+                }
+            }
+            return new Cat(id, name);
+        }
+    }
+}
+```
+
+**Add customized model:**
+
+``` C#
+// Cat.cs
+namespace Azure.Service.Models
+{
+    public partial class Cat
+    {
+        // currently we have to use a full name to ensure this could be replaced
+        void global:Azure.Core.IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName("name"u8);
+            writer.WriteStringValue(Name);
+            // WORKAROUND: server never needs color, remove it in the customization code
+            writer.WriteEndObject();
+        }
+            
+        internal static Cat DeserializeCat(JsonElement element)
+        {
+            string color = default;
+            string name = default;
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("name"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    name = property.Value.GetString();
+                    continue;
+                }
+            }
+            // WORKAROUND: server never sends color, default to black
+            color = "black";
+            return new Cat(name, color);
+        }
+    }
+}
+```
+
+**Generated code after:**
 
 Generated code won't contain the DeserializeCat method and the custom one would be used for deserialization.
```

Comment:
In the customized code, there is `void global:Azure.Core.IUtf8JsonSerializable.Write(Utf8JsonWriter writer)`, so the serialized code will not be generated also. Right?

Reply:
Yes, both serialization and deserialization are overridden by customized code in this case.

Reply:
I tweaked this line to emphasize both `Write` and `DeserializeCat` are gone.